src/Composer/Console/Application.php

```diff
@@ -235,6 +235,9 @@ public function doRun(InputInterface $input, OutputInterface $output): int
                 false === $commandName
                 // list command requires plugin commands to show them
                 || in_array($commandName, ['', 'list', 'help'], true)
+                // autocompletion requires plugin commands but if we are running as root without COMPOSER_ALLOW_SUPERUSER
+                // we'd rather not autocomplete plugins than abort autocompletion entirely, so we avoid loading plugins in this case
+                || $commandName === '_complete' && !$isNonAllowedRoot
```

Comment:
Sorry forgot parenthesis here for completeness

```suggestion
                || ($commandName === '_complete' && !$isNonAllowedRoot)
```